katran/lib/KatranMonitor.cpp

```diff
@@ -49,11 +50,18 @@ KatranMonitor::KatranMonitor(const KatranMonitorConfig& config)
     throw std::runtime_error("none of eventReaders were initialized");
   }
 
-  std::vector<std::shared_ptr<DataWriter>> data_writers;
+std::vector<std::shared_ptr<DataWriter>> data_writers;
   for (int i = 0; i < config_.maxEvents; i++) {
-    std::string fname;
-    folly::toAppend(config_.path, "_", i, &fname);
-    data_writers.push_back(std::make_shared<FileWriter>(fname));
+    if (config_.storage == PcapStorageFormat::FILE) {
+      std::string fname;
+      folly::toAppend(config_.path, "_", i, &fname);
+      data_writers.push_back(std::make_shared<FileWriter>(fname));
+    } else {
+      // PcapStorageFormat::IOBuf
+      buffers_.emplace_back(folly::IOBuf::create(config_.bufferSize));
+      data_writers.push_back(
```

Comment:
Likewise, data_writers are emplaced on the last index. Later we try to pick appropriate writer using eventId. 
One idea that can make it simpler - if we start with a struct that is associated with an event and owns both the buffer and data_writers. Instead of bufferes_ and data_writers it maintains list of events.

Reply:
eventId is returned from XDP forwarding plane as index. there is no way to return "event" (as object) from there.